### ISSUE TITLE
.shortcuts cannot be a simlink

### ISSUE THREAD

**Issue by harshad1** [termux/termux-widget#57]
**Problem description**

I used to be able to use a simlink for `.shortcuts`. In the latest version it has to be a directory.

**Steps to reproduce**

1. Make simlink from `.shortcuts -> ./foo/bar`
2. Widget is empty

**Expected behavior**

Termux shows all items in `./foo/bar`

**Additional information**

* Termux application version: 0.117
* Termux widget version: 0.13 (f droid)
* Android OS version: 11
* Device model: Pixel 3 XL

**Comment by agnostic-apollo** [comment]
That is expected behaviour. Check Changelog for `v0.13.0` at https://github.com/termux/termux-widget/releases

Create a symlink under `~/.shortcuts` directory and place `foo` under `~/.termux`.

**Comment by RalfWerner** [comment]
@harshad1 as discussed [here ff](https://gitter.im/termux/dev?at=6150713b7cd57813a8e799cb) the _"canonical"_ restriction (for tasker) will _not_ changed back for widget and I desribe a [new one](https://github.com/ralfwerner/integrated-process/issues/4)

**Comment by harshad1** [comment]
Thanks for pointing me in the right direction.

Unfortunately, this change has broken my workflow. I use simlinks for both ~/.termux and ~/.shortcuts (both point to a folder on my sdcard which is synced with FolderSync).

Will need to figure out a new system.

**Comment by harshad1** [comment]
> this had to be done

Np. Though could you point me at any discussions of why it was required? Would be interesting to read.

> You can use `rsync` command 

Ah good suggestion. My current workaround is less elegant :p. I have the following in a shortcut

```bash
rm -rf ~/.termux || true
rm -rf ~/.shortcuts || true
cp -r ~/configuration/app_files/termux ~/.termux
cp -r ~/.termux/shortcuts ~/.shortcuts
chmod 700 -R ~/.termux/shortcuts
chmod 700 -R ~/.termux/tasker
```

**Comment by agnostic-apollo** [comment]
It's a restriction for termux plugins that can be controlled by external apps. Same applies to `termux-tasker` for almost a year, it even requires [additional permissions](https://github.com/termux/termux-tasker#Setup-Instructions). It's just being enforced now for `termux-widget`.

> I have the following in a shortcut

Yeah, the other way works too.

**Comment by nate2014jatc** [comment]
Can this be manually overridden?
This breaks how I've historically used the widget.

**Comment by RalfWerner** [comment]
>Can this be manually overridden? This breaks how I've historically used the widget.

Not really, except you use the version V0.12! I also deal with my administration as the _canonical_ restriction is not **negotiable**

**Comment by nate2014jatc** [comment]
> > Can this be manually overridden? This breaks how I've historically used the widget.
> 
> Not really, except you use the version V0.12! I also deal with my administration as the _canonical_ restriction is not **negotiable**

I'm not certain I understand your meaning in the second part of your statement, could you please rephrase?

**Comment by RalfWerner** [comment]
see [above](https://github.com/termux/termux-widget/issues/57#issuecomment-929080498) and links

**Comment by agnostic-apollo** [comment]
>Can this be manually overridden?

You can create the symlinks as discussed above or compile `termux-widget` yourself with the restriction removed. You can also create wrapper scripts under `~/.termux` or `~/.shortcuts` to the actual scripts if the actual scripts are outside those directories.

**Comment by agnostic-apollo** [comment]
Check termux/termux-app#2595 for why this change was enforced.

**Comment by amb3r-dev** [comment]
~/.shortcut/* are the only files in my whole setup that I can't just symlink from my dotfiles repo via GNU stow...
Making a bash script to generate shims in that folder is doable, but inconvenient.

**Comment by amb3r-dev** [comment]
Alright, here's what I came up with.
https://github.com/amb3r-dev/dotfiles/blob/master/termux/.local/bin/termux-shortcuts.sh
This script allows you to define all of your shortcuts as bash functions in a single file, and generate shims for each.
The shims are smart enough to delete themselves if their functions have been removed from `termux-shortcuts.sh`.
**_Note that this will overwrite existing scripts if they have the same names as functions you put in this script._**
